olive/common/utils.py

```diff
@@ -196,3 +214,21 @@ def get_attr(module, attr, fail_on_not_found=False):
                 logger.warning(not_found_message)
                 return None
     return module
+
+
+def find_submodules(module, submodule_types, full_name=False):
+    """Find all submodules of a given type in a module.
+
+    :param module: module to search.
+    :param submodule_type: type of submodule to search for. Can be a single type or a tuple of types.
+    :param full_name: if True, return full name of submodule. Otherwise, return last part of submodule name.
+    :return: list of submodules names.
+    """
+    submodules = set()
+    for name, submodule in module.named_modules():
+        if isinstance(submodule, submodule_types):
+            if full_name:
+                submodules.add(name)
+            else:
+                submodules.add(name.split(".")[-1])
+    return list(submodules)
```

Comment:
yield is better than the list?

Reply:
This returns a list of unique values so I am not sure if `yield` works better. Also, it is currently only used as a list for the qlora pass and conversion pass. Both expect a list.

Reply:
Will address this in a follow up PR as discussed offline